src/ripple/protocol/impl/STCurrency.cpp

```diff
@@ -71,6 +73,12 @@ STCurrency::isEquivalent(const STBase& t) const
 bool
 STCurrency::isDefault() const
 {
+    if (auto const rules = getCurrentTransactionRules())
```

Comment:
This changes semantics of this call. Post-amendment `STCurrency` and `STIssue` will not have the default value. Is this really the right behavior?

Reply:
https://github.com/XRPLF/rippled/issues/4965#issuecomment-2079688630

here is what Ed said about default value

Reply:
Is `isDefault()` used for the metadata insertion only? If not then is returning always `false` consistent with other uses?

Reply:
Yes i believe isDefault() is only used for metadata